app/managers/security_manager/index.ts

```diff
@@ -0,0 +1,477 @@
+            }
+        }
+    }
+
+    /**
+     * Handles app state changes to prompt authentication when resuming from background.
+     */
+    onAppStateChange = async (appState: AppStateStatus) => {
+        if (this.isAuthenticationHandledByEmm()) {
+            return;
+        }
+
+        const isActive = appState === 'active';
+        const isBackground = appState === 'background';
+
+        if (isActive && this.previousAppState === 'background') {
+            if (this.activeServer) {
+                const config = this.getServerConfig(this.activeServer);
+                if (config && config.Biometrics && isMainActivity()) {
+                    const authExpired = this.backgroundSince > 0 && (Date.now() - this.backgroundSince) >= toMilliseconds({minutes: 5});
+                    if (authExpired) {
+                        const isJailbroken = await this.isDeviceJailbroken(this.activeServer);
+                        if (!isJailbroken) {
+                            await this.authenticateWithBiometrics(this.activeServer);
+                        }
+                    }
+                    this.backgroundSince = 0;
+                }
+            }
+        } else if (isBackground) {
+            this.backgroundSince = Date.now();
+        }
+
+        this.previousAppState = appState;
+    };
+
+    /**
+     * Checks if EMM is already enabled and setup
+     * to handle biometric / passcode authentication.
+     */
+    isAuthenticationHandledByEmm = () => {
+        return ManagedApp.enabled && ManagedApp.inAppPinCode;
+    };
+
+    /**
+     * Checks if EMM is already enabled and setup
+     * to handle jailbreak protection.
+     */
+    isJalbreakProtectionHandledByEmm = () => {
+        return ManagedApp.enabled && ManagedApp.cacheConfig?.jailbreakProtection === 'true';
+    };
+
+    /**
+     * Checks if EMM is already enabled and setup
+     * to handle screenshot protection.
+     */
+    isScreenshotProtectionHandledByEmm = () => {
+        return ManagedApp.enabled && ManagedApp.cacheConfig?.blurApplicationScreen === 'true';
+    };
+
+    /**
+     * Get the configuration of a server to prevent screenshots.
+     */
+    isScreenCapturePrevented = (server: string) => {
+        const config = this.getServerConfig(server);
+        if (!config) {
+            return false;
+        }
+
+        return config.PreventScreenCapture == null ? false : config.PreventScreenCapture;
+    };
+
+    /**
+     * Checks if the device is Jailbroken or Rooted.
+     */
+    isDeviceJailbroken = async (server: string, siteName?: string) => {
+        if (this.isJalbreakProtectionHandledByEmm()) {
+            return false;
+        }
+
+        const config = this.getServerConfig(server);
+        if (!config && !siteName) {
+            return false;
+        }
+
+        const locale = DEFAULT_LOCALE;
+        const translations = getTranslations(locale);
+        if (config?.JailbreakProtection || siteName) {
```

Comment:
I don't know the reason behind this, but if `JailbreakProtection` is disabled, why do we still check for it?

Reply:
see above comment